libraries/chain/webassembly/runtimes/eos-vm-oc/code_cache.cpp

```diff
@@ -156,13 +157,17 @@ const code_descriptor* const code_cache_async::get_descriptor_for_code(const dig
       it->second = false;
       return nullptr;
    }
-   if(_queued_compiles.find(ct) != _queued_compiles.end()) {
+   if(auto it = _queued_compiles.get<by_hash>().find(boost::make_tuple(std::ref(code_id), vm_version)); it != _queued_compiles.get<by_hash>().end()) {
+      _queued_compiles.relocate(_queued_compiles.begin(), _queued_compiles.project<0>(it));
```

Comment:
Doesn't this eliminate the strict prioritization of system contracts? imo that's a more important behavior to uphold than trying to prioritize based on MRU. Maybe just remove this `relocate()` which will then make it FIFO, with the exception that system contracts get `push_front()`ed to the front of the line.

Being FIFO is still better than it was previously.

Reply:
Removed